src/rust/engine/process_execution/docker/src/docker_tests.rs

```
@@ -81,6 +112,20 @@ fn platform_for_tests() -> Result<Platform, String> {
     })
 }
 
+async fn assert_container_not_exists(docker: &Docker, container_id: &str) -> Result<(), String> {
```

Maybe rename the "assert" part since in Rust tests assertions usually panic (e.g., `assert_eq!`) to fail the test instead of returning a `Result`? Or maybe consider just calling `panic` if the assertion fails?

Also, you should probably add a `#[must_use = "message"]` annotation to make sure this helper's result is used. `#[track_caller]` would also be useful.

---

```rust
error: `#[track_caller]` on async functions is a no-op
```

I think there's a feature gate to enable some experimental behavior - should I use or just ignore for now?

I added the `must_use` annotation but looks like it's a [nightly Rust feature](https://doc.rust-lang.org/std/hint/fn.must_use.html) only and this code is built against stable Rust.